lib/meson.build

```diff
@@ -0,0 +1,46 @@
+
+lib_include_dir = [include_directories('.')]
+
+libvfio_user_cflags = cc.get_supported_arguments([
+  '-fvisibility=hidden',
+])
+
+libvfio_user_sources = [
+  'dma.c',
+  'irq.c',
+  'libvfio-user.c',
+  'migration.c',
+  'pci.c',
+  'pci_caps.c',
+  'tran.c',
+  'tran_sock.c',
+]
+
+if opt_tran_pipe
+  libvfio_user_sources += ['tran_pipe.c']
+  libvfio_user_cflags += ['-DWITH_TRAN_PIPE']
+endif
+
+libvfio_user_deps = [
+  json_c_dep,
+]
+
+libvfio_user = library(
+  'vfio-user',
+  sources: libvfio_user_sources,
+  c_args: libvfio_user_cflags + common_cflags,
+  dependencies: libvfio_user_deps,
+  include_directories: public_include_dir,
+  soversion: 0,
```

Comment:
could we please re-instate the old comment that we're not yet providing ABI so this doesn't get changed?

Reply:
Sure, will so